src/Tribe/Linked_Posts.php

```diff
@@ -959,6 +940,85 @@ public function render_meta_box_sections( $event ) {
 	}
 
 	/**
+	 * A several-step process that prints the "Create or Find {Linked Post Type Name}" labels.
+	 *
+	 * Numerous steps and caveats are covered in this method so that we can make these labels, which
+	 * are rather important, as translation-friendly as possible.
+	 *
+	 * @since TBD
+	 *
+	 * @param object $post_type The linked post type whose label is being rendered.
+	 * @param boolean $creation_enabled Whether the current user can create post types. If false, they can only add existing ones.
+	 *
+	 * @return string
+	 */
+	public function get_create_or_find_labels( $post_type, $creation_enabled  ) {
+
+		$plural_name             = $this->linked_post_types[ $post_type ]['name'];
+		$singular_name           = ! empty( $this->linked_post_types[ $post_type ]['singular_name'] ) ? $this->linked_post_types[ $post_type ]['singular_name'] : $plural_name;
+		$singular_name_lowercase = ! empty( $this->linked_post_types[ $post_type ]['singular_name_lowercase'] ) ? $this->linked_post_types[ $post_type ]['singular_name_lowercase'] : $singular_name;
+
+		// First, determine what indefinite article we should use for the post type (important for English and several other languages).
+		$indefinite_article = _x( 'a', 'Indefinite article for the phrase "Find a {post type name}. Will be replaced with "an" if the {post type name} starts with a vowel.', 'the-events-calendar' );
+
+		$post_type_starts_with = substr( $singular_name, 0, 1 );
+		$post_type_starts_with = strtolower( $post_type_starts_with );
+		$english_vowels        = array( 'a', 'e', 'i', 'o', 'u' );
+
+		if ( in_array( $post_type_starts_with, $english_vowels ) ) {
+			$indefinite_article = _x( 'an', 'Indefinite article for the phrase "Find a {post type name}" when the {post type name} starts with a vowel, e.g. "Find an Organizer".', 'the-events-calendar' );
+		}
+
+		// Here we render the main label string. The "core" linked post types (venue and organizer) are explicitly named to make
+		// translation a bit easier for the many languages where the words *around* the post type name may need to be different
+		// based on the specific post type name. For non-"core" post types, we just dynamically populate the post type name.
+		switch ( $post_type ) {
+
+			// Organizers
+			case Tribe__Events__Organizer::POSTTYPE :
+
+				if ( tribe_is_organizer_label_customized() ) {
+					$label = sprintf( esc_attr_x( 'Find %1$s %2$s', '"Find an Organizer", but when the word "Organizer" is customized to something else.', 'the-events-calendar' ), $indefinite_article, $singular_name );
```

Comment:
Not sure the translation, escaping and substitution are in the correct order here. As it stands, we  escape the translation (good!) and then we apply the substitutions. 

However, if `$label` is indeed going to be used in an attribute that won't protect us - here and elsewhere we'd instead need to do things in this sort of order:

```php
$label = esc_attr( sprintf( _x( 'Foo %s', 'domain' ), $bar ) );
```

Reply:
good catch, fixed in all cases!